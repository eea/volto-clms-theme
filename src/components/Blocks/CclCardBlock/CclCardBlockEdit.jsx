import React from 'react';
import CclCard from '@eea/volto-clms-theme/components/CclCard/CclCard';

import {
  SidebarPortal,
  BlocksToolbar,
  Icon,
  BlocksForm,
} from '@plone/volto/components'; // BlocksForm, Icon,
import { CardBlockSchema } from './CardBlockSchema';
import InlineForm from '@plone/volto/components/manage/Form/InlineForm';

const CclCardBlockEdit = (props) => {

  const { 
          block, 
          data, 
          onChangeBlock, 
          pathname, 
          selected, 
          manage 
        } = props;
  // console.log(props)
  let card = {
    "product": data.title,
    "description": data.description,
    "image": {
      "scales": {
        "icon": {
          "download":"https://eu-copernicus.github.io/copernicus-component-library/assets/images/image_placeholder.jpg",
        },
        "large": {
          "download":"https://eu-copernicus.github.io/copernicus-component-library/assets/images/image_placeholder.jpg",
        },
        "listing": {
          "download":"https://eu-copernicus.github.io/copernicus-component-library/assets/images/image_placeholder.jpg",
        },
        "mini": {
          "download":"https://eu-copernicus.github.io/copernicus-component-library/assets/images/image_placeholder.jpg",
        },
        "preview": {
          "download":"https://eu-copernicus.github.io/copernicus-component-library/assets/images/image_placeholder.jpg",
        },
        "thumb": {
          "download":"https://eu-copernicus.github.io/copernicus-component-library/assets/images/image_placeholder.jpg",
        },
        "tile": {
          "download":"https://eu-copernicus.github.io/copernicus-component-library/assets/images/image_placeholder.jpg",
        },
      },
    },
    "url": data.url,
  };
  // console.log(data)
  
  return (
    <>


      <div
      onClick={() => {
        props.setSidebarTab(1);
      }}
      aria-hidden="true">
        <CclCard type={data.cardStyle || "line"} card={card} />
      </div>

      <SidebarPortal selected={selected}>
        <InlineForm
          schema={CardBlockSchema()}
          title="Product card block"
          onChangeField={(id, value) => {
            onChangeBlock(block, {
              ...data,
              [id]: value,
            });
          }}
          formData={data}
        />
      </SidebarPortal>
    </>
  );
};

export default CclCardBlockEdit;
