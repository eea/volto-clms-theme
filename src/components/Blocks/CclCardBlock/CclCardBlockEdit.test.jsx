import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import { MemoryRouter } from 'react-router-dom';

import CclCardBlockEdit from './CclCardBlockEdit';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// import { shallow } from 'enzyme';
import { mount } from 'enzyme';

Enzyme.configure({ adapter: new Adapter() });

global.__SERVER__ = true; // eslint-disable-line no-underscore-dangle

const mockStore = configureStore();

describe('CclCardBlockEdit', () => {
  it('CclCardBlockEdit block clicks', () => {
    const store = mockStore({
      content: {
        create: {},
        data: {},
      },
      intl: {
        locale: 'en',
        messages: {},
      },
    });
    const data = {
      title: 'example title',
      description: 'example description',
      image: [
        {
          scales: [
            {
              icon: {
                download:
                  'https://eu-copernicus.github.io/copernicus-component-library/assets/images/image_placeholder.jpg',
              },
              large: {
                download:
                  'https://eu-copernicus.github.io/copernicus-component-library/assets/images/image_placeholder.jpg',
              },
              listing: {
                download:
                  'https://eu-copernicus.github.io/copernicus-component-library/assets/images/image_placeholder.jpg',
              },
              mini: {
                download:
                  'https://eu-copernicus.github.io/copernicus-component-library/assets/images/image_placeholder.jpg',
              },
              preview: {
                download:
                  'https://eu-copernicus.github.io/copernicus-component-library/assets/images/image_placeholder.jpg',
              },
              thumb: {
                download:
                  'https://eu-copernicus.github.io/copernicus-component-library/assets/images/image_placeholder.jpg',
              },
              tile: {
                download:
                  'https://eu-copernicus.github.io/copernicus-component-library/assets/images/image_placeholder.jpg',
              },
            },
          ],
        },
      ],
      url: 'https://www.google.com',
    };
    // const component = shallow(
    const component = mount(
      <Provider store={store}>
        <MemoryRouter>
          <CclCardBlockEdit
            data={data}
            block="1234"
            onChangeBlock={() => {}}
            setSidebarTab={() => {}}
          />
        </MemoryRouter>
      </Provider>,
    );
    const legend = component.find('.ccl-block-editor-header legend');
    legend.simulate('click');
    expect(legend).toBeDefined();
  });
  it('renders a CclCardBlockEdit block component', () => {
    const store = mockStore({
      content: {
        create: {},
        data: {},
      },
      intl: {
        locale: 'en',
        messages: {},
      },
    });
    const data = {
      title: 'example title',
      description: 'example description',
      image: [
        {
          scales: [
            {
              icon: {
                download:
                  'https://eu-copernicus.github.io/copernicus-component-library/assets/images/image_placeholder.jpg',
              },
              large: {
                download:
                  'https://eu-copernicus.github.io/copernicus-component-library/assets/images/image_placeholder.jpg',
              },
              listing: {
                download:
                  'https://eu-copernicus.github.io/copernicus-component-library/assets/images/image_placeholder.jpg',
              },
              mini: {
                download:
                  'https://eu-copernicus.github.io/copernicus-component-library/assets/images/image_placeholder.jpg',
              },
              preview: {
                download:
                  'https://eu-copernicus.github.io/copernicus-component-library/assets/images/image_placeholder.jpg',
              },
              thumb: {
                download:
                  'https://eu-copernicus.github.io/copernicus-component-library/assets/images/image_placeholder.jpg',
              },
              tile: {
                download:
                  'https://eu-copernicus.github.io/copernicus-component-library/assets/images/image_placeholder.jpg',
              },
            },
          ],
        },
      ],
      url: 'https://www.google.com',
    };
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <CclCardBlockEdit
            data={data}
            block="1234"
            onChangeBlock={() => {}}
            onSelectBlock={() => {}}
            onChangeField={() => {}}
            setSidebarTab={() => {}}
          />
        </MemoryRouter>
      </Provider>,
    );

    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
