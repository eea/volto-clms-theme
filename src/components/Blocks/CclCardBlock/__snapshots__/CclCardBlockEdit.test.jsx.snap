// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CclCardBlockEdit renders a CclCardBlockEdit block component 1`] = `
Array [
  <div
    className="ccl-block-editor-header"
  >
    <legend
      aria-hidden="true"
      onClick={[Function]}
    />
  </div>,
  <div
    className="card-line"
  >
    <div
      className="card-image"
    >
      <img
        alt="Placeholder"
        src="ccl-thumbnail-placeholder.jpg"
      />
    </div>
    <div
      className="card-text"
    >
      <div
        className="card-title"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Card default title
        </a>
      </div>
      <div
        className="card-description"
      >
        example description
      </div>
    </div>
  </div>,
]
`;
